### PR TITLE
merge into devel the commits from llvm branch without llvm-specific stuff

### PR DESCRIPTION
**Pull request by adriweb**

### CODE REVIEW

**Comment by runer112** [comment]
Some overall comments:

- Why did we change away from OTB? If I'm the only one that this bothers, I'll suck it up, but if this was just a Mateo being Mateo thing and others don't like it either, please speak up.
- Why are generated files from convpng (convimg?) committed? They're build artifacts.

File: examples/graphx/floodfill/src/main.c [hunk 1]
```diff
@@ -26,7 +26,7 @@ int main(void)
     gfx_Begin();
 
     /* Use the default palettes, this is already set by gfx_Begin */
-    gfx_SetDefaultPalette();
+    gfx_SetDefaultPalette(gfx_8bpp);
```
Comment (runer112) [inline examples/graphx/floodfill/src/main.c]: This line should just be removed, it does nothing in this context.

Comment (adriweb) [reply examples/graphx/floodfill/src/main.c]: done

**Comment by adriweb** [comment]
So, the toolchain builds , and so far from running the built examples, everything seems to be OK except examples/combined/appvar_tilemap which just resets the calc (the autotest failed as expected)

**Comment by mateoconlechuga** [comment]
what is OTB

**Comment by adriweb** [comment]
> what is OTB

A brace style: https://softwareengineering.stackexchange.com/a/99557

**Comment by adriweb** [comment]
Cleaned up the last commits, Travis and Appveyor are happy now too.
I believe this is mergeable into devel.

**Pull request merged**